module-rt/src/bootstrap.rs

```diff
@@ -43,35 +44,53 @@ impl ExportingServicePool {
     }
 }
 
+/// This service trait is compatible with FoundryModule, but slightly changed
+/// to share the Port
+#[remote_trait_object_macro::service]
+trait FoundryModuleInternal: Service {
+    fn initialize(&mut self, arg: &[u8], exports: &[(String, Vec<u8>)]);
+    fn create_port(&mut self, name: &str) -> SRwLock<dyn Port>;
+    fn debug(&mut self, arg: &[u8]) -> Vec<u8>;
+    fn shutdown(&mut self);
+}
+
 struct ModuleContext<T: UserModule> {
     user_context: Option<Arc<Mutex<T>>>,
     exporting_service_pool: Arc<Mutex<ExportingServicePool>>,
+    ports: HashMap<String, Arc<RwLock<ModulePort<T>>>>,
```

Comment:
How `ports` is related to "Fix termination logic" commit?

Reply:
To do this - https://github.com/CodeChain-io/module-rt/pull/4/files#diff-847a1b740457bf10416ff38c2bd69a0aR90